src/devtools/devtoolsAtom.ts

```diff
@@ -0,0 +1,142 @@
+import { PrimitiveAtom, atom } from 'jotai'
+
+type Config = {
+  instanceID?: number
+  name?: string
+  serialize?: boolean
+  actionCreators?: any
+  latency?: number
+  predicate?: any
+  autoPause?: boolean
+}
+
+type Message = {
+  type: string
+  payload?: any
+  state?: any
+}
+
+type ConnectionResult = {
+  subscribe: (dispatch: any) => () => void
+  unsubscribe: () => void
+  send: (action: string, state: any) => void
+  init: (state: any) => void
+  error: (payload: any) => void
+}
+
+type Extension = {
+  connect: (options?: Config) => ConnectionResult
+}
+
+type Options = { name?: string }
+
+const getExtension = (): Extension => {
+  try {
+    return (window as any).__REDUX_DEVTOOLS_EXTENSION__ as Extension
+  } catch (e) {
+    throw new Error('Please install/enable Redux devtools extension')
+  }
+}
+
+export function devtoolsAtom<Value>(
+  anAtom: PrimitiveAtom<Value>,
+  options?: Options
+): PrimitiveAtom<Value> {
+  const isWriteable = !!anAtom.write
+  const atomName = options?.name || anAtom.debugLabel || anAtom.toString()
+
+  const extension = getExtension()
+  const devtools = extension.connect({ name: atomName })
+
+  let isTimeTraveling = false
```

Comment:
This is outside atom, which is shared among multiple instances. Please use `refAtom`. Refer `selectAtom`.